File: beers.rb
# encoding: utf-8

module Beer
  EMOJI = "🍺 "
  DOUBLE_EMOJI = "🍻 "
  ERROR = "🚨  🚔  🚔  🚔  🚨 "
end

class Fixnum
  def beers
    Beer::EMOJI * self
  end

  def double_beers
    Beer::DOUBLE_EMOJI * self
  end

  def pack
    if self == 6
      print_pack(2, 3)
    elsif self == 12
      print_pack(2, 6)
    elsif self == 24
      print_pack(4, 6)
    else
      raise BeerError.new
    end
  end

  alias beer beers

private

  def print_pack(rows, cols)
    (cols.beers + "\n") * rows
  end
end

class BeerError < RuntimeError
  def self.class
    Class.new do
      def to_s
        Beer::ERROR
      end
    end
  end

  def to_s
    Beer::ERROR
  end
end

class String
  def compact
    (length / 4).double_beers + ((length % 4) / 2).beers
  end

  def inspect
    "\n" + to_s
  end
end

class Random
  def self.beers(num = nil)
    # if you know the upper limit, you'll always want the maximum
    # number of beers, anyway
    if num
      num.ceil - 1
    else
      rand(100)
    end.beers
  end
end

class Time
  def beer?
    true
  end
end

puts "Loading BEER FRAMEWORK"
sleep 2
puts "... loaded!"

File: beers.js
console.log("Time for a beer run...")

setTimeout(function() {
	String.prototype.repeat = function( num )
	{
	  return new Array( num + 1 ).join( this );
	}

	var BEER = "🍺 ";
	var DOUBLEBEER ="🍻 ";

	Number.prototype.beers = function() {
		return BEER.repeat(this);
	}

	Number.prototype.doubleBeers = function() {
		return DOUBLEBEER.repeat(this);
	}

	Number.prototype.pack = function() {
		function genPack(rows, cols){
			return (cols.beers() + "\n").repeat(rows)
		}
		if (this == 6) { return genPack(2, 3);}
		else if (this == 12) {return genPack(2, 6);}
		else if (this == 18) {return genPack(3, 6);}
		else if (this == 24) {return genPack(4, 6);}
	}

	Math.random.beers = function(num) {
		if (num == undefined) {var num = 10;}
		return BEER.repeat(Math.floor(this() * num));
	}

	Date.now.beer = function(){
		return true;
	}

	console.log('Done!');
}, 2500)
